pkg/operator/resourcesynccontroller/resourcesync_controller.go

```diff
@@ -109,14 +113,21 @@ func (c *ResourceSyncController) SyncConfigMap(destination, source ResourceLocat
 
 	c.syncRuleLock.Lock()
 	defer c.syncRuleLock.Unlock()
-	c.configMapSyncRules[destination] = source
+	c.configMapSyncRules[destination] = ResourceSourceLocation{
+		ResourceLocation: source,
+		SyncedKeys:       sets.NewString(keys...),
+	}
 
 	// make sure the new rule is picked up
 	c.syncCtx.Queue().Add(c.syncCtx.QueueKey())
 	return nil
 }
 
 func (c *ResourceSyncController) SyncSecret(destination, source ResourceLocation) error {
+	return c.SyncSecretWithKeys(destination, source)
+}
+
+func (c *ResourceSyncController) SyncSecretWithKeys(destination, source ResourceLocation, keys ...string) error {
```

Comment:
so this is the method you will be using in the operator?

Reply:
yes